builder/tencentcloud/cvm/common.go

```diff
@@ -50,54 +52,86 @@ func WaitForInstance(client *cvm.Client, instanceId string, status string, timeo
 }
 
 // WaitForImageReady wait for image reaches statue
-func WaitForImageReady(client *cvm.Client, imageName string, status string, timeout int) error {
-	ctx := context.TODO()
-	req := cvm.NewDescribeImagesRequest()
-	FILTER_IMAGE_NAME := "image-name"
-	req.Filters = []*cvm.Filter{
-		{
-			Name:   &FILTER_IMAGE_NAME,
-			Values: []*string{&imageName},
-		},
-	}
-
+func WaitForImageReady(ctx context.Context, client *cvm.Client, imageName string, status string, timeout int) error {
 	for {
-		var resp *cvm.DescribeImagesResponse
-		err := Retry(ctx, func(ctx context.Context) error {
-			var e error
-			resp, e = client.DescribeImages(req)
-			return e
-		})
+		image, err := GetImageByName(ctx, client, imageName)
```

Comment:
Here for retries you could use something like:

https://github.com/hashicorp/packer/blob/39c25b2c661a2565f2506476ea292ac81507dd36/builder/amazon/common/helper_funcs.go#L54-L65

Reply:
Sure! I have a helper function defined here:
https://github.com/hashicorp/packer/blob/39c25b2c661a2565f2506476ea292ac81507dd36/builder/tencentcloud/cvm/common.go#L116

Reply:
Ah I didn't open it ! 👍  LGTM then :D

Reply:
Thanks anyway :)